lib/Serializer.js

```diff
@@ -0,0 +1,118 @@
+			this._defaultSerializer = name;
+		}
+	}
+
+	remove(name) {
+		validateName(name);
+		if (Object.prototype.hasOwnProperty.call(this._serializers, name)) {
+			delete this._serializers[name];
+		}
+		if (this._defaultSerializer === name) {
+			this._defaultSerializer = "_default";
+		}
+	}
+
+	_serializeMany(documentsArray = [], nameOrOptions) {
+		documentsArray = cleanAndValidateDocumentsArray(documentsArray);
+		return documentsArray.map(doc => doc.serialize(nameOrOptions));
+	}
+
+	_serialize(document, nameOrOptions = this._defaultSerializer) {
+		const inputType = typeof nameOrOptions;
+		let isArray = Array.isArray(nameOrOptions);
+		let options;
+
+		if (inputType === "string") {
+			options = this._serializers[nameOrOptions];
+		} else if (isArray || inputType === "object") {
+			options = nameOrOptions;
+		}
+
+		try {
+			validateOptions(options);
+			isArray = Array.isArray(options);
+			if (isArray) {
+				return includeHandler(document, options, {});
+			}
+
+			let serialized = {};
+			if (options.include) {
+				serialized = includeHandler(document, options.include, serialized);
+			}
+			if (options.exclude) {
+				if (!options.include) {
+					serialized = {...document};
+				}
+				serialized = excludeHandler(document, options.exclude, serialized);
+			}
+			if (options.modify && typeof options.modify === "function") {
+				if (!options.include && !options.exclude) {
+					serialized = {...document};
+				}
+				serialized = options.modify(serialized, document);
+			}
+			return serialized;
+		} catch (error) {
+			// Failing quietly and defaulting to dumping the whole object may not be safest idea, lest we expose sensitive data.
+			return {};
+		}
+	}
+}
+
+const includeHandler = (document, includeRules, serialized) =>
+	includeRules.reduce((_serialized, key) => {
+		if (Object.prototype.hasOwnProperty.call(document, key)) {
+			_serialized[key] = document[key];
+		}
+		return _serialized;
+	}, serialized);
+
+const excludeHandler = (document, excludeRules, serialized) =>
+	excludeRules.reduce((_serialized, key) => {
+		if (Object.prototype.hasOwnProperty.call(_serialized, key)) {
+			delete _serialized[key];
+		}
+		return _serialized;
+	}, serialized);
+
+const validateName = name => {
+	if (!name || typeof name !== "string") {
+		throw new Error("Field name is required and should be of type string");
+	}
+};
+
+const validateOptions = options => {
+	if (!options || !(Array.isArray(options) || typeof options === "object")) {
+		throw new Error("Field options is required and should be an object or array");
+	}
+};
+
+const cleanAndValidateDocumentsArray = documentsArray => {
+	if (!documentsArray || !Array.isArray(documentsArray)) {
+		throw new Error("documentsArray must be an array of document objects");
+	}
+	return documentsArray.filter(doc => typeof doc.serialize === "function");
+};
+
+const _defaultSerializer = {
+	modify: (s, o) => {
+		return {...o};
```

Comment:
I think we should name these a bit better. Not sure if I'm correct on the first one.

```suggestion
	modify: (settings, object) => {
		return {...object};
```

Reply:
modify(serialized, original) would be correct. Not opposed just taking up minimal space for a small default method.

Reply:
@ablankenship10 I do agree, however I think it makes it easier for new contributors to submit PRs with clearer naming. Especially for non obvious methods like this. Generic naming for something like `Array.forEach` or something seems more acceptable. But one of my goals recently is to make the code base as approachable as possible (obviously I failed at that with the craziness going on with binding methods to Model/Document 🤣).

Reply:
Fixed in cabc47162f7fbef33f4bfed7b9999812573b7b3c